### PR TITLE
feat: Complete Trello API abstraction refactor

### PR DESCRIPTION
**Pull request by acoven**
## 🎉 Trello API Abstraction Refactor Complete

This PR completes the major refactoring of Trello API calls into a centralized `class_trel.js` abstraction, following the pattern established by `class_goog.js`.

#### New Architecture:
- **class_trel.js**: Centralized Trello API abstraction with comprehensive error handling and logging
- **Background script**: Handles file uploads with proper multipart/form-data (unchanged)
- **Uploader class**: Coordinates file uploads via Chrome extension messaging (unchanged)
- **Model class**: Business logic and component coordination

#### Key Features:
- **API Key & Token Management**: Centralized Trello.setKey() and authorization handling
- **Authorization Wrapper**: OAuth flow with consistent error handling
- **API Call Wrapper**: All Trello.rest() calls wrapped with logging and error handling
- **Method Abstractions**: High-level methods for common operations
- **Event Integration**: Standardized success/failure events

#### Methods Implemented:
- authorize() / deauthorize() - OAuth flow management
- getUser() / getBoards() / getLists() / getCards() / getMembers() / getLabels() - Data retrieval
- createCard() - Card creation with attachment support
- wrapApiCall() - Core API wrapper with error handling

#### Testing:
- ✅ 23 new tests for class_trel.js
- ✅ All 229 tests passing across 4 test suites
- ✅ Comprehensive error handling and edge case coverage
- ✅ Integration testing with Model class

#### Code Quality:
- ✅ ESLint configuration fixed and all formatting issues resolved
- ✅ Proper separation of concerns maintained
- ✅ No breaking changes to existing functionality
- ✅ Comprehensive documentation updated

#### File Upload Architecture (Preserved):
The refactor correctly maintains the existing file upload architecture:
- **Uploader class** sends Chrome extension messages to background script
- **Background script** handles complex file operations (fetch, blob conversion, FormData, multipart upload)
- **No direct Trello API calls** for file uploads - proper separation maintained

#### Event Flow:
1. Model calls class_trel methods for Trello operations
2. class_trel wraps API calls with error handling and logging
3. File uploads go through Uploader → Background script → Trello API
4. All operations emit standardized events for success/failure

### 📊 Impact:
- **Improved maintainability**: Centralized Trello API management
- **Better error handling**: Comprehensive error handling and logging
- **Cleaner architecture**: Proper separation of concerns
- **Enhanced testing**: Comprehensive test coverage
- **No functionality loss**: All existing features preserved

### 🧪 Testing Results:
```
Test Suites: 4 passed, 4 total
Tests:       229 passed, 229 total
Snapshots:   0 total
Time:        1.268 s
```

### 📚 Documentation:
- Updated docs/refactor_trello.md with completion status
- Comprehensive architecture documentation
- Clear separation of responsibilities

---

**Ready for review and merge!** 🚀

### CODE REVIEW

File: chrome_manifest_v3/class_model.js [hunk 1]
```diff
@@ -386,13 +414,8 @@ class Model {
       return;
     }
```
Comment (acoven) [inline chrome_manifest_v3/class_model.js:415]: 🤖 @cursoragent: ✅ Fixed it! Commit [45fb03a](https://github.com/appliedmedia/gmail-2-trello/commit/45fb03a)

**Review by acoven — COMMENTED** [review]
🤖 @cursoragent: ✅ Fixed it! Commit [45fb03a](https://github.com/appliedmedia/gmail-2-trello/commit/45fb03a)

**Review by acoven — COMMENTED** [review]
🤖 @cursoragent: ✅ Fixed it! Commit [45fb03a](https://github.com/appliedmedia/gmail-2-trello/commit/45fb03a)

File: test/test_class_trel.js [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+      },
+      trelloApiKey: 'test-api-key',
+    };
+
+    // Mock the namespace if not already defined
+    global.G2T = global.G2T || {};
+    // Use Jest's require with proper module resolution
+    require('../chrome_manifest_v3/class_trel.js');
+
+    // Create a fresh Trel instance for each test
```
Comment (acoven) [inline test/test_class_trel.js:68]: 🤖 @cursoragent: ✅ Fixed it! Commit [45fb03a](https://github.com/appliedmedia/gmail-2-trello/commit/45fb03a)

**Review by acoven — COMMENTED** [review]
🤖 @cursoragent: ✅ Fixed it! Commit [45fb03a](https://github.com/appliedmedia/gmail-2-trello/commit/45fb03a)

File: test/test_class_trel.js [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+      },
+      trelloApiKey: 'test-api-key',
+    };
+
+    // Load and evaluate Trel class using vm module for security
+    const fs = require('fs');
+    const vm = require('vm');
+
+    const trelCode = fs.readFileSync(
+      'chrome_manifest_v3/class_trel.js',
```
Comment (acoven) [inline test/test_class_trel.js:68]: 🤖 @cursoragent: ✅ Fixed it! Commit [a61f235](https://github.com/appliedmedia/gmail-2-trello/commit/a61f235) - Used vm module for secure code execution

**Review by acoven — COMMENTED** [review]
🤖 @cursoragent: ✅ Fixed it! Commit [a61f235](https://github.com/appliedmedia/gmail-2-trello/commit/a61f235) - Used vm module for secure code execution
